src/App.js

```diff
@@ -14,11 +15,101 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    sortedBy: '',
+    isReversed: false,
+    listVisible: false,
+    goods: goodsFromServer,
+
+  }
+
+  listIsVisible = () => {
+    this.setState({ listVisible: true });
+  }
+
+  listIsInvisible = () => {
+    this.setState({ listVisible: false });
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  sortAlphabetically = () => {
+    this.setState({ sortedBy: 'name' });
+  }
+
+  sortByLength = () => {
+    this.setState({ sortedBy: 'length' });
+  }
+
+  reset = () => {
+    this.setState({
+      sortedBy: '',
+      isReversed: false,
+    });
+  }
+
+  render() {
+    const { goods, listVisible, isReversed, sortedBy } = this.state;
+    const copyOfGoods = goods.slice();
```

Comment:
```suggestion
    const copyOfGoods = goods.slice();
```
Interesting way to make a copy. It's more common to use the spread operator